### PR TITLE
✅ Test api routes using api.json

### PR DESCRIPTION
**Pull request by astroash**
~~**NB: This will not pass travis tests until TwinePlatform/eslint-config-twine#3 is merged**~~

**Changes:**
- update ts setting to include es7 lib
- add missing descriptions/auth to api.json
- add isImplemented key to api.json
- update linting to tslint via shared config

Relates #27

### CODE REVIEW

File: src/server.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+server.route(router);
+
+const init :()=> Promise <void>=  async () => {
+  try {
+    await server.register({
+      plugin: require('hapi-pino'),
+      options: {
+        prettyPrint: true
+      }
+  })
```
Comment (eliascodes) [inline src/server.ts]: Messed up indent.

File: src/server.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    await server.start();
+    console.log(`Server running at: ${server.info.uri}`);
+  } catch (error) {
+    console.log(error)
+    process.exit(1)
+  }
+};
+export {server, init}
```
Comment (eliascodes) [inline src/server.ts]: Please enable `"files.insertFinalNewline": true` in VS: https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

File: src/__tests__/router.test.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import {server} from '../server'
+const apiJson = require('../../docs/api.json')
+
+type httpMethod = 
+  "GET"
```
Comment (eliascodes) [inline src/__tests__/router.test.ts]: Think somethings happening with the linting, cos these should all be single quote. That's the second thing that hasn't been caught. Can you have a quick look at what's going on?

File: src/__tests__/router.test.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  | "POST"
+  | "PUT"
+  | "DELETE"
+
+type routeTestFixture = {
```
Comment (eliascodes) [inline src/__tests__/router.test.ts]: From what I've seen, convention is that all non-primitive types are capitalised; `RouteTestFixture` and `HttpMethod`.

File: src/__tests__/router.test.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    const fullRoute = Object.entries(apiJson.routes[resource])
+      .reduce((nestedAcc, [nestedRoute, methods]) => {
+        const testObj = Object.entries(methods)
+        .map(([method, routesProps]) => {
+          const url = (resource + nestedRoute).replace(':id', '1')
+          const statusCode = routesProps.isImplemented 
+            ? 200
+            : 404
+          return {
+            testName: `${method} test for route ${url}`,
```
Comment (eliascodes) [inline src/__tests__/router.test.ts]: This can just be `${method} ${url}`. `test for route` is just noise.

File: src/__tests__/router.test.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+            method,
+            statusCode
+          }
+        })
+        return [... nestedAcc, ...testObj]
+      }, [])
+    return [...acc, ...fullRoute]
+}, [])
+
+describe('route testing', () => {
```
Comment (eliascodes) [inline src/__tests__/router.test.ts]: We will be doing other route testing beyond this. What this test suite describes is the `API Specification`.

File: package.json [hunk 7]
```diff
@@ -64,15 +66,12 @@
     "ramda": "^0.25.0"
   },
   "devDependencies": {
+    "@fimbul/bifrost": "^0.9.0",
```
Comment (eliascodes) [inline package.json]: What's this for?

File: package.json [hunk 8]
```diff
@@ -22,8 +22,8 @@
   "main": "src/index.js",
   "scripts": {
     "test": "jest",
-    "lint": "eslint .",
-    "lint:fix": "eslint . --fix",
+    "lint": "tslint --project tsconfig.json",
```
Comment (eliascodes) [inline package.json]: did you install tslint globally? Why not locally?

Comment (astroash) [reply package.json]: It's installed as a dependency of our inaccurately named `eslint-config-twine repo`

File: package.json (hunk 7 above)
Comment (eliascodes) [reply package.json]: @astroash

File: package.json [hunk 9]
```diff
@@ -50,13 +65,12 @@
     "ramda": "^0.25.0"
   },
   "devDependencies": {
-    "eslint": "^4.19.1",
-    "eslint-config-twine": "github:TwinePlatform/eslint-config-twine",
-    "eslint-plugin-import": "^2.12.0",
-    "eslint-plugin-jsx-a11y": "^6.0.3",
-    "eslint-plugin-react": "^7.9.1",
+    "@fimbul/bifrost": "^0.9.0",
```
Comment (eliascodes) [inline package.json]: what's this for?

File: package.json (hunk 7 above)
Comment (astroash) [reply package.json]: Oops, that's a tslint dependency, I'll stick it in the other repo

File: package.json (hunk 9 above)
Comment (astroash) [reply package.json]: It's a tslint dependency, I'll stick it in the other repo

**Comment by astroash** [comment]
@eliascodes this should be good to go

**Pull request merged**